dask_kubernetes/helm.py

```diff
@@ -251,20 +252,28 @@ def scale(self, n_workers):
         >>> cluster.scale(4)
         >>> cluster
         HelmCluster('tcp://localhost:8786', workers=4, threads=24, memory=24.96 GB)
-
+        >>> cluster.scale(5, worker_group = "high-mem-worker")
+        >>> cluster
+        # TODO: Finish scaling example
         """
-        return self.sync(self._scale, n_workers)
+        return self.sync(self._scale, n_workers, worker_group=worker_group)
 
-    async def _scale(self, n_workers):
-        await self.apps_api.patch_namespaced_deployment(
-            name=f"{self.release_name}-{self.chart_name}{self.worker_name}",
-            namespace=self.namespace,
-            body={
-                "spec": {
-                    "replicas": n_workers,
-                }
-            },
-        )
+    async def _scale(self, n_workers, worker_group=None):
+        deployment = f"{self.release_name}-{self.chart_name}{self.worker_name}"
+        if worker_group:
+            deployment += f"-{worker_group}"
+        try:
+            await self.apps_api.patch_namespaced_deployment(
+                name=deployment,
+                namespace=self.namespace,
+                body={
+                    "spec": {
+                        "replicas": n_workers,
+                    }
+                },
+            )
+        except kubernetes.client.exceptions.ApiValueError:
+            raise ValueError(f"No such worker group {worker_group}")
```

Comment:
What happens here if this exception is raised and `worker_group` is not set?

We should probably check if `worker_group` is `None` and if so reraise the exception.